src/manifold/src/boolean_result.cpp

```diff
@@ -567,7 +567,7 @@ void CreateProperties(Manifold::Impl &outR, const Vec<TriRef> &refPQ,
           }
           if (uvw[j] == 0) edge = j;
         }
-        if (edge >= 0) {
+        if (edge >= 0) {  // This misses some edges
```

Comment:
This is the root cause of the problem: when an edge is collapsed (before this step) you can end up with verts that refer (through refPQ) to a triangle that they are now outside of, since they were merged with their neighboring triangle but the references were not updated. This would be fine, except here I'm assuming I can tell when a vert is on a retained edge by checking its barycentric coordinates - but I miss these ones. So I end up assigning a key with only the `meshID` and `vert`, which isn't enough when the other side of the retained edge might have different properties.